jf_agent/config_file_reader.py

```diff
@@ -380,6 +425,79 @@ def get_ingest_config(config: ValidatedConfig, creds, endpoint_jira_info: dict)
             feature_flags={},
         )
 
+    git_configs: List[JFIngestGitConfig] = []
+    is_mult_git_mode = len(config.git_configs) > 1
+    for agent_git_config in config.git_configs:
+        agent_git_config: GitConfig = agent_git_config
+
+        if is_mult_git_mode:
+            instance_slug = agent_git_config.git_instance_slug
+            endpoint_git_instance_info = endpoint_git_instances_info.get(instance_slug)
+            instance_creds = creds.git_instance_to_creds.get(instance_slug)
+        else:
+            # If there's only one git config set, then this is "single git" mode.
+            # The instance slug is likely not to be provided
+            endpoint_git_instance_info = list(endpoint_git_instances_info.values())[0]
```

Comment:
Q: Is this a result of having a "legacy" mode of sending things (non-GQL/ADO), as well as the new code? Are there any failure modes we're concerned about here?

Reply:
Or, now re-reading, is this not a requirement if the custom only provides a single git instance in their `config.yaml`?

Reply:
This logic is used in a few places around the Agent: https://github.com/Jellyfish-AI/jf_agent/blob/08cfab5d49cd7f69d743519184aa9c31a8553b8f/jf_agent/main.py#L723